File: FEB19A/CHEFING.cpp
/*
@author: Ilias Khan
====

Link to the problem: 
    https://www.codechef.com/FEB19A/problems/CHEFING/
Link to my profile:
    https://www.codechef.com/users/iliaskhan
*/


#include <iostream>
using namespace std;
int main() {
	int t;
	cin>>t;
	while(t--)
	{
	    int n;
	    cin>>n;
	    string str[n];
	    int arr[26]={0};
	 
	    for(int i=0;i<n;i++)
	    {
	      
	        cin>>str[i];
	        for(int j=0;j<str[i].size();j++)
	        {
	            arr[str[i][j]-97]= arr[str[i][j]-97]==i?arr[str[i][j]-97]+1:arr[str[i][j]-97];
	        }
	    }
	    int count=0;
	    for(int i=0;i<26;i++)
	    {
	        if(arr[i]==n)
	            count++;
	    }
	    cout<<count<<endl;
	}
	return 0;
}

File: Codechef-2019/FEB19B/CHEFING.py
'''
Link to the problem:
https://www.codechef.com/FEB19B/problems/CHEFING/
Link to my profile:
https://www.codechef.com/users/manasa28
'''

def removeDupWithoutOrder(str):
  return "".join(set(str))

t=input()
for i in range(t):
  n=input()
  l=[]
  count=0
  for j in range(n):
    s=raw_input()
    a=removeDupWithoutOrder(s)
    l.append(a)
  lst2 = sorted(l, key=len)
  x=len(lst2[0])
  for j in range(x):
    flag=0
    for k in range(n):
      if lst2[0][j] in lst2[k]:
        flag=flag+1
    if(flag==n):
      count=count+1
  print count
